testhelper/tests.go

```diff
@@ -432,3 +438,12 @@ func (options *TestOptions) runTest() (string, error) {
 
 	return output, err
 }
+
+// RunTestConsistencyWithOutput Runs consistency checks and returns output of terraform apply
+func (options *TestOptions) RunTestConsistencyWithOutput() (*terraform.PlanStruct, map[string]interface{}, error) {
+	options.testSetup()
+	result, err := options.runTestConsistency()
+	output := terraform.OutputAll(options.Testing, options.TerraformOptions)
+	options.testTearDown()
```

Comment:
inside of your new runTestConsistency() you are also performing a tear down if there was an error, so having it again in the public functions may cause an issue if that happens.

Another approach you can take: have the private `runTestConsistency()` perform all the steps, setup/runTest/get outputs/runTestPlan/check plan/teardown. Basically the whole process. And have it return three outputs, the PlanStruct, the output map, and error. The two public functions then just have the one private function call they make, with only difference being how many of those three outputs they return.

Reply:
Sure, thank you for your valuable input here @toddgiguere 
I have made the above changes.